packages/server/lib/controllers/ratelimit.middleware.ts

```diff
@@ -0,0 +1,70 @@
+import type { Request, Response, NextFunction } from 'express';
+import { createClient } from 'redis';
+import { RateLimiterRedis, RateLimiterMemory, RateLimiterRes } from 'rate-limiter-flexible';
+import { getAccount, getRedisUrl } from '@nangohq/shared';
+import { logger } from '@nangohq/shared';
+
+const rateLimiter = await (async () => {
+    const opts = {
+        keyPrefix: 'middleware',
+        points: 1200,
+        duration: 60,
```

Comment:
I picked a pretty high limit global limit. The goal is for now to prevent an "infinite" number of requests to overload the server when users misbehave. Let me know what you think

Reply:
no need to change anything, just thinking out loud. 
when I set a ratelimiter I always try to put a large window because it allows a few burst but not a constant one, so you don't end up locking the system constantly.
- in a 60s window you can burst 1200 http call every 60s
- in a 600s window you can burst once every 600s, but if you reach the limit sooner you are blocked for a long period of time

In other word, an attacker/bad script would only be block for a few seconds now.

Reply:
Setting the right limit is tricky indeed. A larger window with the same limit also means the avg rate per seconds is lower. 
I am sure we will be tweaking the logic to handle more scenarii in the future but for now I would sleep better at night knowing that beyond a certain threshold, request will be discarded quickly and not hit the rest of the system, especially the db